compositor_common/src/scene/validation.rs

```diff
@@ -71,22 +21,22 @@ impl SceneSpec {
             .map(|node| (&node.node_id, node))
             .collect();
 
-        self.validate_input_pads_are_defined_on_node(&defined_node_ids)?;
-        self.validate_input_pads_are_defined_on_output(&defined_node_ids)?;
-        self.validate_outputs_registered(registered_outputs)?;
-        self.validate_node_ids_uniqueness(defined_node_ids_iter, registered_inputs)?;
-        self.validate_cycles(&transform_nodes)?;
-        self.validate_nodes_are_used(&transform_nodes)?;
-        self.validate_builtin_transformations()?;
+        Self::validate_input_pads_are_defined_on_node(&self.nodes, &defined_node_ids)?;
```

Comment:
the fact that validation methods sometimes used arguments and sometimes self, was hiding complexity. I would even consider removing those functions from `impl` and making them regular unassociated functions.